include/deal.II/lac/arpack_solver.h

```diff
@@ -267,31 +316,41 @@ void ArpackSolver::solve (const MatrixType1                  &/*system_matrix*/,
                           std::vector<VectorType>            &eigenvectors,
                           const unsigned int                  n_eigenvalues)
 {
-  //inside the routines of ARPACK the
-  //values change magically, so store
-  //them here
-
-  const unsigned int n = eigenvectors[0].size();
-  const unsigned int n_inside_arpack = eigenvectors[0].size();
+  // Problem size
+  int n = eigenvectors[0].size();
 
   // Number of eigenvalues for arpack
-  const unsigned int nev = (n_eigenvalues == 0) ? eigenvalues.size() : n_eigenvalues;
-  AssertIndexRange(eigenvalues.size()-1, nev);
-  /*
-  if(n < 0 || nev <0 || p < 0 || maxit < 0 )
-       std:cout << "All input parameters have to be positive.\n";
-       */
+  int nev = (n_eigenvalues == 0) ? eigenvalues.size() : n_eigenvalues;
+
+  // check input sizes
+  if (additional_data.symmetric)
+    {
+      Assert (n_eigenvalues <= eigenvectors.size(),
```

Comment:
nev? 

Since `n_eigenvalues` is an additional optional parameter to specify fewer eigenpairs than the size of vectors, i think it should not be used anywhere but in definition of `nev`.